### ISSUE TITLE
CMake iconv: both test cases passed

### ISSUE THREAD

**Issue by waarmond** [doxygen/doxygen#10716]
**Describe the bug**
When I try to build the ArchLinux [build recipe](https://gitlab.archlinux.org/archlinux/packaging/packages/doxygen/-/raw/main/PKGBUILD?ref_type=heads) but use git main instead of the last tagged release [FindIconv.cmake](https://github.com/doxygen/doxygen/blob/master/cmake/FindIconv.cmake) can't be configured:
```
CMake Error at cmake/FindIconv.cmake:132 (MESSAGE):
  Unable to determine iconv() signature - both test cases passed!
```
My libiconv version: 1.17

**To Reproduce**
Paraphrased version:
```
cmake . \
    -DCMAKE_BUILD_TYPE:STRING=None \
    -DCMAKE_INSTALL_PREFIX:PATH=/usr \
    -Dbuild_wizard:BOOL=ON \
    -Duse_sys_spdlog:BOOL=ON \
    -Duse_sys_sqlite3:BOOL=ON \
    -Duse_libclang:BOOL=ON
cmake --build . --verbose
```

**Expected behavior**
CMake test case for iconv shall be unique.
Succeeding ICONV_ACCEPTS_CONST_INPUT & ICONV_ACCEPTS_NONCONST_INPUT should be not an issue.

**Version**
git current
```
LSB Version:	n/a
Distributor ID:	Arch
Description:	Arch Linux
Release:	rolling
```

same issue as https://github.com/doxygen/doxygen/issues/6786 https://trac.macports.org/ticket/51371

**Comment by doxygen** [comment]
@waarmond Does it work if you remove the `cmake/FindIconv.cmake` file that ships with doxygen and rely on the `FindIconv.cmake` that ships with cmake instead?

**Comment by albert-github** [comment]
I did a quick test under Cygwin and Windows by removing the FindIConv.cmake as provided by doxygen.

- Cygwin: 
  Errors:
  ```
  /usr/lib/gcc/x86_64-pc-cygwin/11/../../../../x86_64-pc-cygwin/bin/ld: ../lib/libdoxycfg.a(portable_c.c.o):portable_c.c:(.text+0x1): undefined reference to `libiconv_open'
  /usr/lib/gcc/x86_64-pc-cygwin/11/../../../../x86_64-pc-cygwin/bin/ld: ../lib/libdoxycfg.a(portable_c.c.o):portable_c.c:(.text+0x11): undefined reference to `libiconv'
  /usr/lib/gcc/x86_64-pc-cygwin/11/../../../../x86_64-pc-cygwin/bin/ld: ../lib/libdoxycfg.a(portable_c.c.o):portable_c.c:(.text+0x21): undefined reference to `libiconv_close'
  ```
- Windows:
  Errors:
  ```
  CMake Error at D:/Program Files/CMake/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
    Could NOT find Iconv (missing: Iconv_LIBRARY Iconv_INCLUDE_DIR)
  Call Stack (most recent call first):
    D:/Program Files/CMake/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
    D:/Program Files/CMake/share/cmake-3.22/Modules/FindIconv.cmake:165 (find_package_handle_standard_args)
    CMakeLists.txt:170 (find_package)
  ```

**Comment by Croydon** [comment]
#10888 might have fixed the issues here

**Comment by albert-github** [comment]
@waarmond  Please test

Code (#10888) has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
